libs/api_common/windows_platform_common.cpp

```diff
@@ -136,10 +428,734 @@ get_attach_type_windows(const std::string& section)
 _Ret_maybenull_z_ const char*
 get_attach_type_name(_In_ const ebpf_attach_type_t* attach_type)
 {
-    // TODO: (Issue #223) Read the registry to fetch attach types.
-    auto it = windows_section_names.find(*attach_type);
-    if (it != windows_section_names.end())
-        return it->second;
+    for (const ebpf_section_definition_t& t : _windows_section_definitions) {
+        if (IsEqualGUID(*t.attach_type, *attach_type)) {
+            return t.section_prefix;
+        }
+    }
+
+    return nullptr;
+}
+
+static ebpf_result_t
+_load_section_data_information(HKEY section_data_key, _In_ const wchar_t* section_name) noexcept
+{
+    int32_t status;
+    ebpf_result_t result = EBPF_SUCCESS;
+    HKEY section_info_key = nullptr;
+    ebpf_program_type_t* program_type = nullptr;
+    ebpf_attach_type_t* attach_type = nullptr;
+    ebpf_section_definition_t section_definition;
+    bpf_prog_type_t bpf_program_type;
+    bpf_attach_type_t bpf_attach_type;
+    char* section_prefix = nullptr;
+
+    try {
+        status = open_registry_key(section_data_key, section_name, KEY_READ, &section_info_key);
+        if (status != ERROR_SUCCESS) {
+            // Registry path is not present.
+            result = EBPF_FILE_NOT_FOUND;
+            goto Exit;
+        }
+
+        program_type = (ebpf_program_type_t*)ebpf_allocate(sizeof(ebpf_program_type_t));
+        if (program_type == nullptr) {
+            result = EBPF_NO_MEMORY;
+            goto Exit;
+        }
+
+        attach_type = (ebpf_attach_type_t*)ebpf_allocate(sizeof(ebpf_attach_type_t));
+        if (attach_type == nullptr) {
+            result = EBPF_NO_MEMORY;
+            goto Exit;
+        }
+
+        // Read program type.
+        status = read_registry_value_binary(
+            section_info_key, EBPF_SECTION_DATA_PROGRAM_TYPE, (uint8_t*)program_type, sizeof(ebpf_program_type_t));
+        if (status != ERROR_SUCCESS) {
+            result = win32_error_code_to_ebpf_result(status);
+            goto Exit;
+        }
+
+        // Read attach type.
+        status = read_registry_value_binary(
+            section_info_key, EBPF_SECTION_DATA_ATTACH_TYPE, (uint8_t*)attach_type, sizeof(ebpf_attach_type_t));
+        if (status != ERROR_SUCCESS) {
+            result = win32_error_code_to_ebpf_result(status);
+            goto Exit;
+        }
+
+        // Read bpf program type.
+        status =
+            read_registry_value_dword(section_info_key, EBPF_PROGRAM_DATA_BPF_PROG_TYPE, (uint32_t*)&bpf_program_type);
+        if (status != ERROR_SUCCESS) {
+            bpf_program_type = BPF_PROG_TYPE_UNSPEC;
+            result = EBPF_SUCCESS;
+        }
+
+        // Read bpf attach type.
+        status =
+            read_registry_value_dword(section_info_key, EBPF_PROGRAM_DATA_BPF_PROG_TYPE, (uint32_t*)&bpf_attach_type);
```

Comment:
I would recommend not using "registry" in the function name, so that if it moves to say the filesystem in the future, this code needn't change.